### PR TITLE
service: initial OS service infrastructure

### PR DESCRIPTION
**Pull request by amery**

### CODE REVIEW

File: pkg/service/service.go [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+	d   time.Duration
+	err atomic.Value
+	ss  service.Service
+	p   program
+
+	cmd     cobra.Command
+	prepare func(context.Context, *cobra.Command, []string) error
+	run     func(context.Context, *cobra.Command, []string) error
+
+	initializer []func()
```
Comment (karasz) [inline pkg/service/service.go:31]: Why is this singular and `finalizers` plural? seem sto me that this should be `initializers`

**Review by karasz — CHANGES_REQUESTED** [review]

File: pkg/service/service.go (hunk 1 above)
Comment (amery) [reply pkg/service/service.go]: good catch, thanks

**Comment by amery** [comment]
@karasz amended

**Review by karasz — APPROVED** [review]
LGTM

**Comment by amery** [comment]
`darvaza.org/sidecar/pkg/service@v0.0.1` tagged

**Pull request merged**